runtime/common/src/lib.rs

```diff
@@ -0,0 +1,1649 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+#![allow(clippy::crate_in_macro_def)]
+
+pub mod weights;
+
+#[macro_export]
+macro_rules! decl_common_types {
```

Comment:
Yes, this would cause a circular dependency. At least some of the types declared in that macro need types that are generated during the `construct_runtime!` macro. If we declare them in `common`, then `zeitgeist` and `battery-station` need `common` to construct the runtime, but the types used in the runtime and generated by `construct_runtime!` would require `common` to include `zeitgeist` and `battery-station`.

Other projects often just duplicate them. Means `zeitgeist` and `battery-station` would have huge chunks of identical code (they do now too but it's hidden through the macro).

Reply:
Thanks for explaining. I have no strong opinion on this issue. Both approaches seem very reasonable.

Reply:
TBH. I am not 100% happy with this solution. I was quite crestfallen when I realized that what you proposed is not possible. I am also not happy with code duplication. I like how light the current runtimes look, but I dislike the disadvantages that come with using macros for that. Maybe we find a better approach for that in the future, but I think we can leave it like it is for now.

Reply:
Absolutely, no more changes! It's a perfectly fine approach.